src/line.ts

```diff
@@ -1,83 +1,93 @@
-import * as d3 from 'd3';
-import { DateTime } from 'luxon'
-import { generateCube } from './datagen/generation';
-import { categories, measures } from './data_config';
-import { betweenDates } from './datagen/filters';
+import d3 from 'd3';
+import {DateTime} from 'luxon';
+import {generateCube} from './datagen/generation';
+import {categories, measures} from './data_config';
+import {betweenDates} from './datagen/filters';
 
-interface Data {
+interface Datum {
   date: Date;
   value: number;
 }
 
 const cube = generateCube(categories, measures);
 
-function getData(): Data[] {
+function getData(): Datum[] {
   const endDate = DateTime.local();
-  const startDate = endDate.minus({ day: 30 });
+  const startDate = endDate.minus({day: 30});
 
   const data = cube.getDataFor(
-    ['nthDay'], 
-    ['activeUsers'], 
-    [betweenDates(
-        new Date(startDate.toMillis()), 
-        new Date(endDate.toMillis()))
-        ]
-        );
-  return data.map((datum) => ({
-      date: startDate.plus({ days: datum.categories.get('nthDay') }),
-      value: datum.values.get('activeUsers')!,
+    ['nthDay'],
+    ['activeUsers'],
+    [betweenDates(startDate.toJSDate(), endDate.toJSDate())]
+  );
+  return data.map(datum => ({
+    date: startDate
+      .plus({days: datum.categories.get('nthDay') as number})
+      .toJSDate(),
+    value: datum.values.get('activeUsers')!,
   }));
 }
 
 export async function createLineChart() {
   const height = 500;
   const width = 800;
-  const margin = ({top: 20, right: 30, bottom: 30, left: 40});
+  const margin = {top: 20, right: 30, bottom: 30, left: 40};
 
-  const svg = d3.create("svg")
-      .attr("viewBox", [0, 0, width, height]);
+  const svg = d3.create('svg').attr('viewBox', [0, 0, width, height].join(' '));
 
   const data = getData();
 
-  const xAxis = (g: d3.Selection<SVGGElement>) => g
-      .attr("transform", `translate(0,${height - margin.bottom})`)
-      .call(d3.axisBottom(x).ticks(width / 80).tickSizeOuter(0));
-  const yAxis = (g: d3.Selection<SVGGElement>) => g
-      .attr("transform", `translate(${margin.left},0)`)
+  const xAxis = (g: d3.Selection<SVGGElement, any, any, any>) =>
```

Comment:
```suggestion
  const xAxis = (g: d3.Selection<SVGGElement, Datum[], SVGElement, Datum[]>) =>
```

Reply:
Per `@types/d3-selection`,
```
/**
 * A D3 Selection of elements.
 *
 * The first generic "GElement" refers to the type of the selected element(s).
 * The second generic "Datum" refers to the type of the datum of a selected element(s).
 * The third generic "PElement" refers to the type of the parent element(s) in the D3 selection.
 * The fourth generic "PDatum" refers to the type of the datum of the parent element(s).
 */
```

Shouldn't the 2nd and 4th generic still remain as `any` since the axis element and the svg element don't have a datum bound?

I've looked into [their test code](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/68c3c174697d1bf27af59a0c88d05a8d8eff454f/types/d3-selection/d3-selection-tests.ts#L765), and it seems like they also leave none-bound element as `any`.

Reply:
Actually, I think it makes more sense to have it as `d3.Selection<SVGGElement, undefined, SVGElement, undefined>)` since we never _define_ which datum to bind. How do you think?

Reply:
Or we can just inline `xAxis` and `yAxis` as in 7f1f9df to avoid explicitly using `any`. In this case, `g` on [line 72](https://github.com/googleinterns/guide-doge/pull/7/commits/7f1f9dfb4d921c111eedef970dc01d5ca09a5002#diff-2bcbe349d0874ef3d101f576c00dee35R72) is inferred to be `d3.Selection<SVGGElement, undefined, null, undefined>`. (I can revert the commit if it looks worse.)